front/src/modules/timesStops/helpers/computeMargins.tsx

```diff
@@ -0,0 +1,108 @@
+import type { SimulationResponseSuccess } from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainScheduleV2/utils';
+import {
+  decimalToPercentage,
+  mToMm,
+  milliUnitToBaseUnit,
+  minutePer100km,
+  percentageToDecimal,
+  timeLostFromMin100KmMargin,
+} from 'utils/physics';
+import type { ValueOf } from 'utils/types';
+
+import { formatDigitsAndUnit } from './utils';
+import { MarginUnit, marginRegExValidation } from '../consts';
+
+// We use the previous unit used if it’s not given
+type MarginUnit = ValueOf<typeof MarginUnit>;
+let marginUnit: MarginUnit = MarginUnit.percent;
+
+const defaultReturn = {
+  theoreticalMarginWithUnit: '',
+  calculatedMargin: '',
+  diffMarginsWithUnit: '',
+};
```

Comment:
It's a detail, but don't hesitate to add blank lines to make the code easier to read
```suggestion
};

```

Reply:
done